lmnet/lmnet/networks/keypoint_detection/lm_single_pose_v1.py

```diff
@@ -0,0 +1,218 @@
+        return functools.partial(lmnet_block,
+                                 activation=self.activation,
+                                 custom_getter=self.custom_getter,
+                                 is_training=is_training,
+                                 is_debug=self.is_debug,
+                                 use_bias=False,
+                                 data_format=channels_data_format)
+
+    def _space_to_depth(self, inputs=None, block_size=2, name=''):
+        if self.data_format != 'NHWC':
+            inputs = tf.transpose(inputs, perm=[self.data_format.find(d) for d in 'NHWC'])
+        output = tf.space_to_depth(inputs, block_size=block_size, name=name)
+        if self.data_format != 'NHWC':
+            output = tf.transpose(output, perm=['NHWC'.find(d) for d in self.data_format])
+        return output
+
+    def base(self, images, is_training, *args, **kwargs):
+        channels_data_format = 'channels_last' if self.data_format == 'NHWC' else 'channels_first'
+        _lmnet_block = self._get_lmnet_block(is_training, channels_data_format)
+
+        def connect_block(lateral, up, name="connect_block"):
+            with tf.variable_scope(name_or_scope=name):
+                shape = lateral.get_shape().as_list()
+                up = tf.image.resize_nearest_neighbor(up, size=shape[1:3])
+                out = tf.concat([up, lateral], axis=-1)
+                out = _lmnet_block("block", out, filters=32, kernel_size=3)
+            return out
+
+        self.images = images
+
+        x = _lmnet_block('conv1', images, 32, 1)
+        x = self._space_to_depth(name='space2depth1', inputs=x)
+        x = _lmnet_block('conv2', x, 32, 3)
+        x = _lmnet_block('conv3', x, 32, 3)
+        c1 = _lmnet_block('c1', x, 32, 1)
+
+        x = self._space_to_depth(name='space2depth2', inputs=x)
+        x = _lmnet_block('conv4', x, 32, 3)
+        x = _lmnet_block('conv5', x, 32, 3)
+        x = _lmnet_block('conv6', x, 32, 3)
+        c2 = _lmnet_block('c2', x, 32, 1)
+
+        x = self._space_to_depth(name='space2depth3', inputs=x)
+        x = _lmnet_block('conv7', x, 64, 3)
+        x = _lmnet_block('conv8', x, 64, 3)
+        x = _lmnet_block('conv9', x, 64, 3)
+        c3 = _lmnet_block('c3', x, 32, 1)
+
+        x = self._space_to_depth(name='space2depth4', inputs=x)
+        x = _lmnet_block('conv10', x, 128, 3)
+        x = _lmnet_block('conv11', x, 128, 3)
+        x = _lmnet_block('conv12', x, 128, 3)
+        c4 = _lmnet_block('c4', x, 32, 1)
+
+        x = self._space_to_depth(name='space2depth5', inputs=x)
+        x = _lmnet_block('conv13', x, 256, 3)
+        x = _lmnet_block('conv14', x, 256, 3)
+        x = _lmnet_block('conv15', x, 256, 3)
+        x = _lmnet_block('conv16', x, 256, 3)
+        x = _lmnet_block('conv17', x, 256, 3)
+        c5 = _lmnet_block('c5', x, 32, 1)
+
+        x = connect_block(lateral=c4, up=c5, name="connect_block1")
+
+        if self.stride < 16:
```

Comment:
This nested if looks not good.
Can you replace with `if` and `elif` ?

Reply:
simply removing the nesting would do i think.
```
if s < 16:
    do_something()
if s < 8:
    do_more()
...
```